src/test/java/uk/gov/pay/directdebit/payers/services/PayerServiceTest.java

```diff
@@ -64,10 +67,22 @@ public void setUp() {
     }
 
     @Test
-    public void shouldStoreAPayerWhenReceivingCreatePayerRequest() {
-        service.create(paymentRequestExternalId, gatewayAccount.getExternalId(), createPayerRequest);
+    public void shouldCreateAndStoreAPayerWhenReceivingCreatePayerRequest() {
+        service.createOrUpdatePayer(paymentRequestExternalId, gatewayAccount, createPayerRequest);
         verify(mockedPayerDao).insert(payer);
         verify(mockedTransactionService).payerCreatedFor(transaction);
+        verify(mockedTransactionService, never()).payerEditedFor(transaction);
+    }
+    @Test
+    public void shouldUpdateAndStoreAPayerWhenReceivingCreatePayerRequest_ifAPayerAlreadyExists() {
+        Payer payer = PayerFixture.aPayerFixture()
+                .withName("mr payment").toEntity();
+        when(mockedPayerDao.findByPaymentRequestId(transaction.getPaymentRequest().getId()))
+                .thenReturn(Optional.of(payer));
+        service.createOrUpdatePayer(paymentRequestExternalId, gatewayAccount, createPayerRequest);
+        verify(mockedPayerDao).updatePayerDetails(eq(payer.getId()), any(Payer.class));
```

Comment:
I’m always a bit cautious of `any` matchers. If I’m not mistaken, this will actually be the field `payer`. But maybe you could do something like this to make it really clear:
```java
Payer editedPayer = mock(Payer.class);
when(mockedPayerParser.parse(createPayerRequest, transaction)).thenReturn(editedPayer);
…
verify(mockedPayerDao).updatePayerDetails(payer.getId(), editedPayer);
```

Reply:
Hah, yes, good catch. I didn't think about returning a mock ( :D ) in this case, so simple! Thanks